backend/items/api.py

```
@@ -148,6 +167,79 @@ def check_rate_limit(request):
+    if not user_id:
+        user_id = request.META.get(
+            "HTTP_X_FORWARDED_FOR", request.META.get("REMOTE_ADDR", "unknown")
+        )
+        if "," in user_id:
+            user_id = user_id.split(",")[0].strip()
+
+    try:
+        response = es_agent_stream_rate_limiter.limit(user_id)
+        if not response.allowed:
+            reset_time = response.reset
+            hours_until_reset = (reset_time - datetime.now().timestamp()) / 3600
```

Ruff (0.15.2)

[warning] `datetime.datetime.now()` called without a `tz` argument

(DTZ005)